### ISSUE TITLE
AppSyncGraphQLRequestFactory.buildQuery build wrong query

### ISSUE THREAD

**Issue by lwang-79** [aws-amplify/amplify-android#1814]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

GraphQL API

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>

### Environment information

<details>

```
# Put output below this line
implementation 'com.amplifyframework:core:1.36.1'
implementation 'com.amplifyframework:aws-auth-cognito:1.36.1'
implementation 'com.amplifyframework:aws-api:1.36.1'
implementation 'com.amplifyframework:aws-auth-cognito:1.36.1'

```

</details>

### Please include any relevant guides or documentation you're referencing

https://aws.amazon.com/getting-started/hands-on/build-android-app-amplify/module-four/

### Describe the bug

I tried to follow the tutorial, but I couldn't list the notes items.

After print the query result I found the following error message

'Validation error of type FieldUndefined: Field 'listNoteDatas' in type 'Query' is undefined @ 'listNoteDatas'', locations='[GraphQLLocation{line='2', column='3'}]', path='null', extensions='null'

Checked the query name in AppSync it's 'listNoteData' instead of 'listNoteDatas'.

So I think the Amplify Core and Amplify CLI use different name to build the query. Is there any workaround except manually change the schema in AppSync console?

### Reproduction steps (if applicable)

Follow the tutorial,
Create a NoteData schema,
amplify codegen models
Try to list the items

### Code Snippet

```kotlin
    fun queryNotes() {
        Log.i(TAG, "Querying notes")

        Amplify.API.query(
            ModelQuery.list(NoteData::class.java),
            { response ->
                Log.i(TAG, "Queried")
                Log.i(TAG, response.toString())
                if (response.data != null) {
                    for (noteData in response.data) {
                        Log.i(TAG, noteData.name)
                        // TODO should add all the notes at once instead of one by one (each add triggers a UI refresh)
                        UserData.addNote(UserData.Note.from(noteData))
                    }
                }
            },
            { error -> Log.e(TAG, "Query failure", error) }
        )
    }
```

### Log output

<details>

```
// Put your logs below this line
'Validation error of type FieldUndefined: Field 'listNoteDatas' in type 'Query' is undefined @ 'listNoteDatas'', locations='[GraphQLLocation{line='2', column='3'}]', path='null', extensions='null'

```

</details>

### amplifyconfiguration.json

[amplifyconfiguration.json.zip](https://github.com/aws-amplify/amplify-android/files/9069692/amplifyconfiguration.json.zip)

### GraphQL Schema

<details>

```graphql
type NoteData
@model
@auth (rules: [ { allow: owner } ]) {
    id: ID!
    name: String
    description: String
    image: String
}

```

</details>

**Comment by lwang-79** [comment]
Yes, I have tested with some different nouns. Looks like the library add "s" for all names, but the CLI follows the grammar.

**Comment by lawmicha** [comment]
The https://github.com/aws-amplify/amplify-codegen/pull/255 PR has been closed but I don't think the Android changes have been merged in. I can see in one of the codegen java tests that it's still got `pluralName` instead of splitting it into the two `list` and `sync` plural names (https://github.com/aws-amplify/amplify-codegen/blob/main/packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-java-visitor.test.ts.snap#L26). 

The next steps is to apply the codegen changes, use the closed PR as reference and Swift codegen PR as reference https://github.com/aws-amplify/amplify-codegen/pull/667

@lwang-79 if you are still experience pluralname mismatch, one workaround is to use a different model name that does get plurailzed with `s` such as `Note` instead of `NoteData`
